packages/beacon-node/src/util/graffiti.ts

```diff
@@ -18,10 +18,12 @@ export function getDefaultGraffiti(
   }
 
   if (executionClientVersion != null) {
-    const {code: executionCode, commit: executionCommit} = executionClientVersion;
+    let {code: executionCode, commit: executionCommit} = executionClientVersion;
+
+    executionCommit = executionCommit.startsWith("0x") ? executionCommit.slice(2, 6) : executionCommit.slice(0, 4);
```

Comment:
why is there are 0x prefix in the first place? I think our deserialization is not correct, we also don't serialize it correctly according to https://github.com/ChainSafe/lodestar/issues/7217

Reply:
`commit` has the type `DATA` as per https://github.com/ethereum/execution-apis/blob/main/src/engine/identification.md#clientversionv1

`DATA` type must be prepended by `0x` as per https://github.com/ethereum/execution-apis/blob/main/src/engine/common.md#encoding

Reply:
> we also don't serialize it correctly according to https://github.com/ChainSafe/lodestar/issues/7217

Yea will address it in a separate PR

Reply:
> DATA type must be prepended by 0x as per https://github.com/ethereum/execution-apis/blob/main/src/engine/common.md#encoding

but isn't that the encoding in transit, e.g. see `parentHash` for example, it's bytes internally but will be hex encoded in transit

https://github.com/ChainSafe/lodestar/blob/84e069930156b1fabb2e2cd9261d58fcd6eff6b1/packages/beacon-node/src/execution/engine/types.ts#L217

see data conversion functions
https://github.com/ChainSafe/lodestar/blob/84e069930156b1fabb2e2cd9261d58fcd6eff6b1/packages/beacon-node/src/eth1/provider/utils.ts#L101-L119